routers/web/repo/render.go

```diff
@@ -62,12 +62,14 @@ func RenderFile(ctx *context.Context) {
 		treeLink += "/" + util.PathEscapeSegments(ctx.Repo.TreePath)
 	}
 
-	ctx.Resp.Header().Add("Content-Security-Policy", "frame-src 'self'; sandbox allow-scripts")
+	ctx.Resp.Header().Add("Content-Security-Policy", "frame-src 'self'; sandbox allow-scripts allow-same-origin")
```

Comment:
If the rendered content contains user's javascript, would `allow-same-origin` cause XSS?

Reply:
Sandbox attribute with `allow-scripts allow-same-origin` is a noop and has the same effect as no sandbox attribute. From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe):

> When the embedded document has the same origin as the embedding page, it is strongly discouraged to use both allow-scripts and allow-same-origin, as that lets the embedded document remove the sandbox attribute — making it no more secure than not using the sandbox attribute at all.

Not that I see a big issue here because openapi specs can not contain executable code.

Reply:
But this code is for all renders, not only for OpenAPI.

Reply:
For PDF renderer, it can definitely be an issue as those could maybe contain JS.

Reply:
So if the iframe really needs `allow-same-origin`, we should specifically allow it only for the openapi renderer, otherwise remove the attribute.